daemon/api.go

```diff
@@ -1237,3 +1244,44 @@ func abortChange(c *Command, r *http.Request, user *auth.UserState) Response {
 
 	return SyncResponse(change2changeInfo(chg), nil)
 }
+
+var (
+	postCreateUserUcrednetGetUID = ucrednetGetUID
+	storeUserInfo                = store.UserInfo
+	osutilAddExtraUser           = osutil.AddExtraUser
+)
+
+func postCreateUser(c *Command, r *http.Request, user *auth.UserState) Response {
+	uid, err := postCreateUserUcrednetGetUID(r.RemoteAddr)
+	if err != nil {
+		return BadRequest("cannot get ucrednet uid: %v", err)
+	}
+	if uid != 0 {
+		return BadRequest("cannot use create-user as non-root")
+	}
+
+	var createData struct {
+		EMail string `json:"email"`
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&createData); err != nil {
+		return BadRequest("cannot decode create-user data from request body: %v", err)
+	}
+
+	v, err := storeUserInfo(createData.EMail)
```

Comment:
Best to check for the case of empty email or username so we can provide better error messages for missing data in the payloads.